eclass/unpacker.eclass

```diff
@@ -429,7 +429,21 @@ _unpacker_get_decompressor() {
 	*.lzma|*.xz|*.txz)
 		echo "xz -T$(makeopts_jobs) -dc" ;;
 	*.lz)
-		: ${UNPACKER_LZIP:=$(type -P plzip || type -P pdlzip || type -P lzip)}
+		find_lz_unpacker() {
+			local lz_unpacker
+			local has_version_arg="-b"
+
+			[[ ${EAPI} == 6 ]] && has_version_arg="--host-root"
+
+			for lz_unpacker in \
+				$(has_version ${has_version_arg} ">=app-arch/xz-utils-5.4.0" && echo "xz") \
```

Comment:
If you're doing `has_version` already, there's no point in doing `type -P` as well. Then the whole function can be a lot cleaner.

Reply:
Well, the idea was to drop it later once it's stable, so we don't need to do the slowish has_version in future. But I don't mind converting it all if you want.

Reply:
But then you'd need to practically rewrite this nightmare of a function anyway, wouldn't you?

Reply:
Thinking about it, we won't be able to enforce newer xz-utils in `@system` for a long time (by the way, I think we should consider setting some >= in `packages` to avoid users with e..g ancient GNU Make), so yes, I agree - let me rewrite this now.

Reply:
I tried a few different things and this version I've just pushed looks best IMO. WDYT?